nilearn/plotting/find_cuts.py

```diff
@@ -323,3 +323,127 @@ def find_cut_slices(img, direction='z', n_cuts=7, spacing='auto'):
     cut_coords.sort()
 
     return _transform_cut_coords(cut_coords, direction, affine)
+
+
+def find_parcellation_cut_coords(labels_img, background_label=0, return_label_names=False,
+                                 label_hemisphere='left'):
+    """ Return coordinates of center of mass of 3D parcellation atlas
+
+    Parameters
+    ----------
+    labels_img: 3D Nifti1Image
+        A brain parcellation atlas with specific mask labels for each
+        parcellated region.
+
+    background_label: int, optional (default 0)
+        Label value used in labels_img to represent background.
+
+    return_label_names: bool, optional (default False)
+        Returns list of labels
+
+    label_hemisphere: 'left' or 'right', optional (default 'left')
+        Choice of hemisphere to compute label center coords for.
+        Applies only in cases where atlas labels are laterlized.
+        Eg. Yeo or Harvard Oxford atlas.
+
+    Returns
+    -------
+    coords: numpy.ndarray of shape (n_labels, 3)
+        Label regions cut coordinates in image space (mm).
+
+    labels_list: list
+        Label region.
+
+    See Also
+    --------
+    nilearn.plotting.find_probabilistic_atlas_cut_coords : For coordinates
+        extraction on probabilistic atlases (4D) (Eg. MSDL atlas)
+    """
+    # Grab data and affine
+    labels_img = reorder_img(check_niimg_3d(labels_img))
+    labels_data = labels_img.get_data()
+    labels_affine = labels_img.affine
+
+    # Grab number of unique values in 3d image
+    unique_labels = set(np.unique(labels_data)) - set([background_label])
+
+    # Loop over parcellation labels, grab center of mass and dump into coords
+    # list
+    coord_list = []
+    label_list = []
+
+    for cur_label in unique_labels:
+        cur_img = labels_data == cur_label
+
+        # Below inspecting whether the current atlas label is (a)laterlized,
+        # (b)spans hemipsheres, or
+        # (c)is disconnected and present in both hemispheres
+
+        # Grab hemispheres separately
+        x, y, z = coord_transform(0, 0, 0, np.linalg.inv(labels_affine))
+        left_hemi = labels_img.get_data().copy() == cur_label
+        right_hemi = labels_img.get_data().copy() == cur_label
+        left_hemi[int(x):] = 0
+        right_hemi[:int(x)] = 0
+
+        # Case 1: One label in one hemisphere
+        if np.all(left_hemi == False) or np.all(right_hemi == False):
+            pass
+        # Case 2: Connected component spanning both hemispheres
+        # elif np.any(right_hemi[int(x)]) and np.any(left_hemi[int(x-1)]):
+        #    if not np.any(right_hemi[int(x-10)]) and
+        #    np.any(left_hemi[int(x-10)]):
+        #        pass
+        # Case 3: Two Connected component in both hemispheres
```

Comment:
The code above needs to be cleaned up: the if clause should be removed and the else should be turned to a "if not ()" the clause above.